scripts/multisig_tx.py

```
@@ -0,0 +1,317 @@
+    safe_tx = safe.get_safe_tx_by_safe_tx_hash(safe_tx_hash)
+
+    if log.prompt_yes_no("Sign transaction?"):
+        signature = safe.sign_with_frame(safe_tx)
+        if log.prompt_yes_no("Post signature?"):
+            safe.post_signature(safe_tx, signature)
+
+
+def fake_factory() -> None:
+    """Use to deploy factory for testing purpose"""
+    from brownie import history
+
+    config = _read_envs()
+    safe = ApeSafe(config["SAFE_ADDRESS"])
+
+    lido_treasury = "0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c"  # some address with LDOs
+    ldo = Contract.from_explorer("0x5A98FcBEA516Cf06857215779Fd812CA3beF1B32")
+    vesting = VestingEscrow.deploy({"from": safe.address})
+    adapter = VotingAdapter.deploy(
+        "0xffffffffffffffffffffffffffffffffffffffff",
+        "0xffffffffffffffffffffffffffffffffffffffff",
+        "0xffffffffffffffffffffffffffffffffffffffff",
+        "0xffffffffffffffffffffffffffffffffffffffff",
+        {"from": "0x1111111111111111111111111111111111111111"},
+    )
+    factory = VestingEscrowFactory.deploy(
+        vesting,
+        LDO_ADDRESS,
+        "0xffffffffffffffffffffffffffffffffffffffff",
+        "0x0000000000000000000000000000000000000000",
+        adapter,
+        {"from": "0x1111111111111111111111111111111111111111"},
+    )
+    ldo.transfer(safe.address, 100_000 * 10**18, {"from": lido_treasury})
+    ldo.approve(factory, 100_000 * 10**18, {"from": safe.address})
+    history.clear()  # to avoid these transactions to occur in multisend
+
+    log.info(f"{factory.address=}")
+    input("Press ENTER to exit...")
+
+
+def _preview_and_check_tx(safe: ApeSafe, safe_tx: SafeTx, params_list: Sequence["VestingParams"], is_prod=False):
+    vestings_sum = sum(p.amount for p in params_list)
+    starting_balance = _ldo_balance(safe.address)
+
+    log.info("Preview transaction")
+    # do not reset chain in testing to keep the fake factory and other contracts intact
+    tx = safe.preview(safe_tx, reset=is_prod)
```

Prompt after preview?

---

It will be right after the balance check, which we want to fail fast.